src/shipit_static_analysis/tests/conftest.py

```diff
@@ -326,3 +328,21 @@ def mock_revision():
     rev = Revision()
     rev.mercurial = 'a6ce14f59749c3388ffae2459327a323b6179ef0'
     return rev
+
+
+@pytest.fixture
+def mock_clang(tmpdir):
+    '''
+    Mock clang binary setup by linking the system wide
+    clang tools into the expected repo sub directory
```

Comment:
Cool trick! But won't the "system wide clang tools" be missing the `mozilla-*` checks? If so, this will probably perma-fail my unit test from #961. (In which case maybe we'll disable the unit test but keep the runtime check and later add an integration test with the real `clang-tidy`.)

Reply:
As #961 just got merged, i'll rebase on top, and update your code to use the correct binary when available. The unit test would indeed fail for `mozilla-*` checks, but it would still be valid during runtime execution.

IMHO we **really** need integration tests

Reply:
Agreed!
@abpostelnicu Do you have any update on the tests? Thanks

Reply:
After looking other ahal’s patch I’m still trying to figure out how we can use that for our needs. Definitely we will have something on witch we can test shortly.
My main concern is about running the tests on try since we are highly dependent on the state of the toolchain from TC. What I’m saying is depending where we run or tests, m-i or m-c we might end up in the situation that toolchain artifacts are not ready for the current revision.
This is because the toolchain have ids based on a hash from several files from our repo. So if one file changes and lands and the toolchain build is not ready we will fail. Maybe I’m looking too far raising this issue but still I wanted to have it mentioned.